test/pylib/util.py

```diff
@@ -124,13 +124,21 @@ async def read_barrier(cql: Session, host: Host):
 # Wait for the given feature to be enabled.
 async def wait_for_feature(feature: str, cql: Session, host: Host, deadline: float) -> None:
     async def feature_is_enabled():
-        rs = await cql.run_async("select value from system.scylla_local where key = 'enabled_features'", host=host)
-        if rs:
-            value = rs[0].value
-            if feature in value:
-                return True
-        return None
+        enabled_features = await get_enabled_features(cql, host)
+        return feature in enabled_features or None
     await wait_for(feature_is_enabled, deadline)
 
 
+async def get_supported_features(cql: Session, host: Host) -> set[str]:
+    """Returns a set of cluster features that a node advertises support for."""
+    rs = await cql.run_async(f"SELECT supported_features FROM system.local WHERE key = 'local'", host=host)
+    return set(rs[0].supported_features.split(","))
+
+
+async def get_enabled_features(cql: Session, host: Host) -> set[str]:
+    """Returns a set of cluster features that a node considers to be enabled."""
+    rs = await cql.run_async(f"SELECT value FROM system.scylla_local WHERE key = 'enabled_features'", host=host)
+    return set(rs[0].value.split(","))
```

Comment:
The code before this patch didn't seem to have the split()... Was it simply wrong?

Reply:
If I understand you correctly, you mean the fact that before it was `string1 in string2` and now it's `string1 in string2.split(",")`?

The code before the patch was wrong, but it was OK as long as no feature name was a substring of another feature name. (Note that in Python `string1 in string2` checks whether `string1` is a substring of `string1`. So before the patch, we would check whether the feature name is a substring of the comma-separated string of features. After the patch, we explicitly construct the list (Python list) of features from the comma-separated string, and check whether the feature is in the list.

Reply:
Right, I forgot that "string in string" also "sort of, sometimes" works. Of course the new version with "string in set" is better.

Reply:
It used to work, but it was less explicit. `str1 in str2` in python is equivalent to "str1 is a substring of str2", so previously the function `wait_for_feature` checked whether the feature name is a _substring_ of the comma-separated string of features.

Reply:
> The code before the patch was wrong, but it was OK as long as no feature name was a substring of another feature name.

Right, I missed that possibility. So the previous version wasn't entirely correct after all.